tests/conditional_compilation/test_cc.py

```diff
@@ -80,39 +81,47 @@ def test_minimized_pkg(test_id, model, openvino_root_dir, artifacts):  # pylint:
 @pytest.mark.dependency(depends=["cc_collect", "minimized_pkg"])
 def test_infer(test_id, model, artifacts):
     """Test inference with conditional compiled binaries."""
-    out = artifacts / test_id
+    out = artifacts / test_id / test_id
     minimized_pkg = out / "install_pkg"
+
     return_code, output = run_infer(model, f"{out}_cc.npz", minimized_pkg)
     assert return_code == 0, f"Command exited with non-zero status {return_code}:\n {output}"
 
 
 @pytest.mark.dependency(depends=["cc_collect", "minimized_pkg"])
 def test_verify(test_id, model, openvino_ref, artifacts, tolerance=1e-6):  # pylint: disable=too-many-arguments
     """Test verifying that inference results are equal."""
-    out = artifacts / test_id
+    out = artifacts / test_id / test_id
+
     minimized_pkg = out / "install_pkg"
     out_file = f"{out}.npz"
     out_file_cc = f"{out}_cc.npz"
     return_code, output = run_infer(model, out_file, openvino_ref)
     assert return_code == 0, f"Command exited with non-zero status {return_code}:\n {output}"
     return_code, output = run_infer(model, out_file_cc, minimized_pkg)
     assert return_code == 0, f"Command exited with non-zero status {return_code}:\n {output}"
-    reference_results = dict(np.load(out_file))
-    inference_results = dict(np.load(out_file_cc))
-    assert sorted(reference_results.keys()) == sorted(
-        inference_results.keys()
-    ), "Results have different number of layers"
-    for layer in reference_results.keys():
-        assert np.allclose(
-            reference_results[layer], inference_results[layer], tolerance
-        ), "Reference and inference results differ"
+    reference_results = dict(np.load(out_file, allow_pickle=True))
+    inference_results = dict(np.load(out_file_cc, allow_pickle=True))
+    for file in reference_results.keys():
+        assert (reference_results[file].size ==
+                inference_results[file].size
+                ), "Reference and inference has different number of model"
+        for model_number in range(reference_results[file].size):
+            assert sorted(reference_results[file][model_number][0].keys()) == sorted(
```

Comment:
can't figure out why it's required to get 0th index - let's reorganize way of storing data to prevent extra wraps